site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js

```diff
@@ -0,0 +1,290 @@
+import { Formik, Form } from "formik";
+import { Form as SemanticForm, Button, Modal } from "semantic-ui-react";
+import { TextField, TextAreaField, ToggleField } from "react-invenio-forms";
+import CategoryDropdown from "./CategoryDropdown";
+import * as Yup from "yup";
+import axios from "axios";
+import Dropzone from "react-dropzone";
+import FileTable from "./FileTable";
+import _remove from "lodash/remove";
+import { humanReadableBytes } from "react-invenio-deposit";
+
+// TODO translations
+// TODO implement error notification
+// TODO implement recaptcha
+// TODO required fields are highlighted by the component and not Formik
+
+const apiConfig = {
+    withCredentials: true,
+    xsrfCookieName: "csrftoken",
+    xsrfHeaderName: "X-CSRFToken",
+    headers: {
+        "Accept": "application/json",
+        'Content-Type': 'multipart/form-data',
+    },
+};
+
+const http = axios.create(apiConfig);
+
+function formikToFormData(values) {
+    const formData = new FormData();
+    // Append each key/value pair to formData
+    Object.keys(values).forEach((key) => {
+        let currentValue = values[key];
+        if (Array.isArray(currentValue)) {
+            currentValue.forEach((val) => {
+                formData.append(key, val);
+            })
+        } else {
+            formData.append(key, currentValue);
+        }
+    })
+
+    return formData;
+}
+
+
+const FileUploader = ({ dropzoneParams, maxFileSize, name, currentFiles, handleDelete }) => {
+    return (
+        <Dropzone {...dropzoneParams}>
+            {({ getRootProps, getInputProps }) => {
+                const inpProps = {
+                    ...getInputProps(),
+                    // Display the dropzone input, otherwise it's hidden by default.
+                    style: {
+                        'display': 'block'
+                    },
+                    name: name
+                }
+                return (
+                    <div {...getRootProps()}>
+                        <input {...inpProps} />
+                        <label className="helptext mt-0">
+                            {`Optional. Max attachments size:`} {humanReadableBytes(maxFileSize, false)}
+                        </label>
+                        {currentFiles.length > 0
+                            &&
+                            <FileTable
+                                handleDelete={handleDelete}
+                                filesList={currentFiles}
+                            />
+                        }
+                    </div>
+                )
+            }}
+        </Dropzone>
+    );
+};
+
+FileUploader.propTypes = {
+    dropzoneParams: PropTypes.object.isRequired,
+    maxFileSize: PropTypes.number.isRequired,
+    name: PropTypes.string.isRequired,
+    currentFiles: PropTypes.array.isRequired
+};
+
+class SupportForm extends Component {
+
+    constructor(props) {
+        super(props);
+        this.FormSchema = Yup.object({
```

Comment:
would it make sense to use backend validation like in the deposit form? I find yup syntax extremely unreadable

Reply:
Managed to use it, however it required some changes in the backend (e.g. adding a error handler for `marshmallow.ValidationError`.